pkg/custom-provider/series_registry_test.go

```diff
@@ -117,17 +117,60 @@ var seriesRegistryTestSeries = [][]prom.Series{
 	},
 }
 
+type myType struct {
+	a int
+	b string
+	m map[string]int
+}
+
+type mapWrapper struct {
+	item map[string]int
+}
+
+func (o *myType) Mutate(newMap mapWrapper) {
+	o.a = 2
+	o.b = "two"
+	o.m = newMap.item
+}
+
+func TestWeirdStuff(t *testing.T) {
```

Comment:
what's this?

Reply:
Haha. This was just a bit of code I was playing with to try to understand why the compiler was telling me some type was not suitable as the key for a `map`. It wound up not mattering, since I didn't need to key on the type anyway, but during a period where everything was starting off as a clone of `basicSeriesRegistry` the compiler was unhappy. The type, specifically, was `ExternalMetricInfo`. I thought, at first, it's because the language wouldn't allow a `map` to be the key to another `map`, but the simple test showed that's obviously not correct.

Reply:
I'd encourage checking the go language specification when you have questions like this.  It's actually quite readable, compared to other languages.  At any rate, map keys must be comparable, as defined here: https://golang.org/ref/spec#Comparison_operators